crates/bevy_pbr/src/render/mesh.wgsl

```diff
@@ -23,14 +27,24 @@ struct VertexOutput {
 [[group(2), binding(0)]]
 var<uniform> mesh: Mesh;
 
+#ifdef SKINNED
+[[group(3), binding(0)]]
```

Comment:
Bind groups are pretty limited in practice. Now that joint matrices are a dynamic uniform, this seems like something that should be merged into the Mesh binding.

Reply:
Hmm this is actually a tough nut to crack. In theory, this is as simple as moving the skinned joint binding into the mesh bind group.

In practice, we would need to re-create each mesh bind group each time the joint buffer gets reallocated. That would require some pretty fundamental changes to How Things Are Done, so it feels out of scope for this pr.

Reply:
Oh wait no! Meshes dont have individual bind groups. I misremembered. They also have a global one, so this should actually be straightforward. I'll throw it together and see how it looks.